runhouse/servers/cluster_servlet.py

```diff
@@ -84,6 +85,16 @@ async def aset_cluster_config(self, cluster_config: Dict[str, Any]):
 
         self.cluster_config = cluster_config
 
+        new_cluster_name = self.cluster_config.get("name", None)
+
+        if self._cluster_name != new_cluster_name:
```

Comment:
confused by what's going on here - why do we need to reset the cluster name if we area already pulling it from the cliuster config above?

Reply:
We are setting a new cluster_config, so we want to verify that the cluster name saved as ClusterServlet property is matching with the cluster_name set in the new cluster_config we set.

Reply:
should we even be allowing updating the name here? i feel like that can lead to all sorts of issues with mismatches between what's saved on the cluster vs in den